src/cgutils.cpp

```diff
@@ -1089,6 +1089,9 @@ static jl_cgval_t emit_getfield_knownidx(const jl_cgval_t &strct, unsigned idx,
         fieldval.gcroot = strct.gcroot;
         return fieldval;
     }
+    else if (isa<UndefValue>(strct.V)) {
+        return jl_cgval_t();
```

Comment:
or more callers need to look at `strct.typ` instead of `exprtype` (since exprtype can be wrong / meaningless in dead code)